announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -1579,6 +1658,28 @@ protected String buildPreviewContext(VelocityPortlet portlet, Context context, R
 		}
 		}
 		
+		Collection annToRoles = state.getTempAnnounceToRoles();
+		String allRolesString = "";
+		if (annToRoles!=null) {
+			Site site = null;
+			int count = 0;
+			try {
+				site = SiteService.getSite(EntityManager.newReference(channelId).getContext());		
+				for (Iterator i = annToRoles.iterator(); i.hasNext();) {
+					if (i != null) {
+						count++;
+						if (count > 1) {
+							allRolesString = allRolesString.concat(", ").concat((String) i.next());
+						} else {
+							allRolesString = (String) i.next();
+						}
+					}
+				}
+				context.put("annToRoles", allRolesString);
```

Comment:
What do you want to do here?

Reply:
This is for the preview, to get the column with the correct values

Reply:
StringUtils.join (annToRoles , ',') ; ?